src/Qowaiv.DomainModel/EventBuffer_TId.cs

```diff
@@ -116,4 +123,20 @@ internal string DebuggerDisplay
         => Version == CommittedVersion
         ? $"Version: {Version}, Aggregate: {AggregateId}"
         : $"Version: {Version} (Committed: {CommittedVersion}), Aggregate: {AggregateId}";
+
+    /// <inheritdoc />
+    int ICollection<object>.Count => Buffer.Count;
```

Comment:
The downside is that is should, under no circumstance be confused with the version. That's why I previously did not implemented IReadOnlyCollection<object>, but IEnumerable<object>. That was the original reason, and a (small) downside.

Now ICollection is implemented, it could be argued, that it implements IReadOnlyCollection too. Still, I'm not sure if I want to make Count public.

What kind of downsides do you see?

Reply:
It would allow you to know beforehand how many elements you will be copying when you invoke `CopyTo` (which you did expose)

Reply:
I'm convinced. That is indeed more or less a necessity.